freeipa/src/main/java/com/sequenceiq/freeipa/service/image/userdata/UserDataService.java

```diff
@@ -63,7 +64,16 @@
     private CachedEnvironmentClientService environmentClientService;
 
     public void createUserData(Long stackId) {
-        Stack stack = stackService.getStackById(stackId);
+        Stack stack = getStack(stackId);
+        createUserData(stack, () -> ccmUserDataService.fetchAndSaveCcmParameters(stack));
+    }
+
+    public void regenerateUserData(Long stackId) {
+        Stack stack = getStack(stackId);
+        createUserData(stack, stack::getCcmParameters);
```

Comment:
could you add 1-1 log line to these methods. we don't have too much log in this class

Reply:
sure